### PR TITLE
[15.0][IMP] survey_contact_generation: link partner

### PR DESCRIPTION
**Pull request by chienandalu**
For some advanced workflows is quite convenient to have the linked input on the generated contacts.

cc @Tecnativa TT47573

please review @pedrobaeza @victoralmau

### CODE REVIEW

File: survey_contact_generation/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Copyright 2024 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    survey_user_input_id = fields.Many2one(comodel_name="survey.user_input")
```
Comment (pedrobaeza) [inline survey_contact_generation/models/res_partner.py]: Shouldn't be the reverse direction? Put the generated partner in the survey_input?

Comment (chienandalu) [reply survey_contact_generation/models/res_partner.py]: Already is, but I wan't also to flag those generated partners so I can easily discriminate them for other workflows (look at the tt task for an example)

Comment (pedrobaeza) [reply survey_contact_generation/models/res_partner.py]: And what about the inverse one2many?

Comment (chienandalu) [reply survey_contact_generation/models/res_partner.py]: I just want to link inputs to partners generated by that input, but the input could have a partner which isn't generated (that's the normal behavior)

Comment (pedrobaeza) [reply survey_contact_generation/models/res_partner.py]: OK, let's try to make it more self-discovering by name:

```suggestion
    generating_survey_user_input_id = fields.Many2one(comodel_name="survey.user_input")
```

Comment (chienandalu) [reply survey_contact_generation/models/res_partner.py]: Done :+1:

**Review by pedrobaeza — APPROVED** [review]

**Review by victoralmau — APPROVED** [review]

**Pull request merged**